### ISSUE TITLE
About the result of dataset

### ISSUE THREAD

**Issue by zxk19981227** [IDEA-Research/OSX#32]
when i visualize the dataset, i meet a problem that even with an empty image, there still exists a human. So what's the problem?
![4](https://user-images.githubusercontent.com/40291547/236666254-fd1bdd5d-3c9b-4a66-a667-0828d6d9a545.jpg)

**Comment by zxk19981227** [comment]
I`
import matplotlib.pyplot as plt
import json
import os
import sys
os.environ["PYOPENGL_PLATFORM"] = "osmesa"
import os.path
import smplx
import torch
from utils.renderer import PyRenderer
import cv2
import numpy as np
import trimesh
import pyrender
from matplotlib import pyplot as plt
from mpl_toolkits.mplot3d import Axes3D
from mpl_toolkits.mplot3d.art3d import Poly3DCollection
from pycocotools.coco import COCO



# smplx annotations file 包含两个部分 cam_param 和 smplx_param
part_name='Olympic'
data=json.load(open(os.path.join(root_path,f'annotations/{part_name}/smplx_annotation.json')))
annotation_json_file=json.load(open(os.path.join(root_path,f'annotations/{part_name}/keypoint_annotation.json')))
for image_id in range(len(annotation_json_file['images']))[:100]:
    image_name=annotation_json_file['images'][image_id]['file_name']
    image_idx=list(data.keys())[image_id]
    # print(data.keys())
    # exit()
    print(data[image_idx].keys())
    # print(data[image_idx]['cam_param'])
    # print(data[image_idx]['smplx_param'])
    # print(data[image_idx]['smplx_param'].keys())
    people_pose=data[image_idx]['smplx_param']
    camera_pose=data[image_idx]['cam_param']
    # for key in people_pose.keys():
    #     print(f"{key}:{np.array(people_pose[key]).shape}")
    smplx_layer=smplx.create('smpl/',model_type='smplx')
    vertices_people=smplx_layer(
        global_orient=torch.Tensor(people_pose['root_pose']).unsqueeze(0),
        body_pose=torch.Tensor(people_pose['body_pose']).view(1,21,3).unsqueeze(0),
        lhand_pose=torch.Tensor(people_pose['lhand_pose']).view(1,15,3),
        rhand_pose=torch.Tensor(people_pose['rhand_pose']).view(1,15,3),
        transl=torch.Tensor(people_pose['trans']).view(1,3),
        betas=torch.tensor(people_pose['shape']).view(1,10),
        expression=torch.Tensor(people_pose['expr']).view(1,10),
        jaw_pose=torch.Tensor(people_pose['jaw_pose']).view(1,1,3)
    )
    vertices=vertices_people.vertices.detach().numpy().squeeze()
    extra_shape=np.ones((10475,1))
    vertices=np.concatenate([vertices,extra_shape],axis=1)
    # print(f"shape for vertices is {vertices.shape}")
    fx,fy=camera_pose['focal']
    cx,cy=camera_pose['princpt']
    origin_image=cv2.imread(os.path.join(f'images/{part_name}/',image_name))
    camera_parameters=np.array([
        [fx,0,cx,0],
        [0,fy,cy,0],
        [0,0,1,0]
    ])
    verts2d=camera_parameters.dot(vertices.transpose(1,0)).transpose(1,0)
    verts2d=verts2d[:,:2]/verts2d[:,2:]
    for joint in verts2d:
        # print(joint)
        cv2.circle(origin_image,center=(int(joint[0]),int(joint[1])),radius=1,color=(0,0,255),thickness=-1)
    cv2.imshow(f'{image_id}.jpg',origin_image)`
i visualize the dataset with up code.

**Comment by linjing7** [comment]
Hi, could you please tell me the image name you show? I think it's because the person bounding box detector has detected a person in this image. This is why we ask the annotator to manually annotate the box and we'll release them in UBody-V2.

**Comment by zxk19981227** [comment]
Thanks for you reply， i only visualize the first 100 images in the movie part. What's more, the images directory doesn't match the path in the annoation path

**Comment by linjing7** [comment]
Hi, according to your code, it seems that what you visualize is the Olympic instead of the Movie scene. Besides, what's the mismatch between that paths?

**Comment by zxk19981227** [comment]
It is true that i'm testing on the Olymapic part. The mismatch is when i use the code convert videos to pictures, images of Movie is placed under `movie/images/Movie_S1` rather than `Movie/Movie_S1`

**Comment by linjing7** [comment]
Okay, there is a small mistake in the name of `Movie` folder. Thanks for your remind and I'll upload the video tomorrow.

**Comment by linjing7** [comment]
@zxk19981227 Hi, we have corrected the name of `Movie` folders. Please download the newer-version videos and annotations from google drive. Besides, the mistake that human exists in the empty frame is because the method you read the annotations is not correct. You can try the following visualization code, please place it in the folder of `OSX/tools/UBody`:

```
import json
import smplx
import torch
import cv2
import numpy as np
import sys
import os
from pycocotools.coco import COCO
import argparse

cur_dir = os.path.dirname(os.path.abspath(__file__))
root_dir = os.path.join(cur_dir, '..', '..')
main_dir = os.path.join(cur_dir, '..', '..', 'main')
sys.path.insert(0, root_dir)
sys.path.insert(0, main_dir)


if __name__ == '__main__':
    parser = argparse.ArgumentParser()
    parser.add_argument('--scene', type=str, default='Fitness')
    args = parser.parse_args()
    root_path = '../../dataset/UBody/'
    human_model_folder = '../../common/utils/human_model_files'
    smplx_annot_path = os.path.join(root_path, f'annotations/{args.scene}/smplx_annotation.json')
    kpt_annot_path = os.path.join(root_path, f'annotations/{args.scene}/keypoint_annotation.json')

    with open(smplx_annot_path) as f:
        smplx_params = json.load(f)
    db = COCO(kpt_annot_path)
    for aid in db.anns.keys():
        ann = db.anns[aid]
        img = db.loadImgs(ann['image_id'])[0]
        img_path = os.path.join(root_path, 'images', args.scene, img['file_name'])
        body_keypoints = np.array(ann['keypoints'], dtype=np.float32).reshape(-1, 3)
        if str(aid) not in smplx_params: continue
        smplx_param = smplx_params[str(aid)]
        people_pose = smplx_param['smplx_param']
        camera_pose = smplx_param['cam_param']
        smplx_layer = smplx.create(human_model_folder, model_type='smplx')
        vertices_people = smplx_layer(
            global_orient=torch.Tensor(people_pose['root_pose']).unsqueeze(0),
            body_pose=torch.Tensor(people_pose['body_pose']).view(1, 21, 3).unsqueeze(0),
            lhand_pose=torch.Tensor(people_pose['lhand_pose']).view(1, 15, 3),
            rhand_pose=torch.Tensor(people_pose['rhand_pose']).view(1, 15, 3),
            transl=torch.Tensor(people_pose['trans']).view(1, 3),
            betas=torch.tensor(people_pose['shape']).view(1, 10),
            expression=torch.Tensor(people_pose['expr']).view(1, 10),
            jaw_pose=torch.Tensor(people_pose['jaw_pose']).view(1, 1, 3)
        )
        vertices = vertices_people.vertices.detach().numpy().squeeze()
        faces = smplx_layer.faces

        extra_shape = np.ones((10475, 1))
        vertices = np.concatenate([vertices, extra_shape], axis=1)
        fx, fy = camera_pose['focal']
        cx, cy = camera_pose['princpt']
        origin_image = cv2.imread(img_path)
        camera_parameters = np.array([
            [fx, 0, cx, 0],
            [0, fy, cy, 0],
            [0, 0, 1, 0]
        ])
        verts2d = camera_parameters.dot(vertices.transpose(1, 0)).transpose(1, 0)
        verts2d = verts2d[:, :2] / verts2d[:, 2:]
        for joint in verts2d:
            cv2.circle(origin_image, center=(int(joint[0]), int(joint[1])), radius=1, color=(0, 0, 255), thickness=-1)
        save_path = img_path.replace('images', 'mesh_vis').replace('png', 'jpg')
        os.makedirs(os.path.dirname(save_path), exist_ok=True)
        cv2.imwrite(save_path, origin_image)
```
